framework/src/org/apache/cordova/CordovaActivity.java

```diff
@@ -115,8 +116,24 @@ public void onCreate(Bundle savedInstanceState) {
             getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                     WindowManager.LayoutParams.FLAG_FULLSCREEN);
         } else if (preferences.getBoolean("Fullscreen", false)) {
-            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+            {
+                setSystemUiVisibilityMode(getWindow());
+                getWindow().getDecorView().setOnSystemUiVisibilityChangeListener(new View.OnSystemUiVisibilityChangeListener() {
```

Comment:
I don't think this listener should be necessary. According to:
https://developer.android.com/training/system-ui/immersive.html

setting IMMERSIVE_STICKY should take care of re-entering fullscreen.